python/sanlock.c

```diff
@@ -1546,22 +1551,22 @@ with -EBUSY error in this case.\n\
 static PyObject *
 py_set_event(PyObject *self __unused, PyObject *args, PyObject *keywds)
 {
-    const char *lockspace;
+    PyObject *lockspace;
     struct sanlk_host_event he = {0};
     uint32_t flags = 0;
     int rv;
 
     static char *kwlist[] = {"lockspace", "host_id", "generation", "event",
                              "data", "flags", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, keywds, "sKKK|KI", kwlist,
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "SKKK|KI", kwlist,
         &lockspace, &he.host_id, &he.generation, &he.event, &he.data,
         &flags)) {
         return NULL;
     }
 
     Py_BEGIN_ALLOW_THREADS
-    rv = sanlock_set_event(lockspace, &he, flags);
+    rv = sanlock_set_event(PyBytes_AsString(lockspace), &he, flags);
     Py_END_ALLOW_THREADS
```

Comment:
We don't have tests for this code, need to add minimal tests.

This patch is rather big, lets defer this work for now. I want to merge the first path, where all the code and tests pass on python 3, and work on better types in python 3 and smarter path handling in the next phase.

Reply:
Okay.